### PR TITLE
content-sources: add client CreateRepository.

### PR DESCRIPTION
**Pull request by ldjebran**
# Description
The CreateRepository the repositories function that will be used to create repository in content-sources.

100% coverage

FIXES: THEEDGE-3208

## Type of change

What is it?

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Documentation update
- [ ] Tests update
- [ ] Refactor

# Checklist:

- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] I run `make pre-commit` to check fmt/vet/lint/test-no-fdo

### CODE REVIEW

File: pkg/clients/repositories/clients_test.go [hunk 1]
```diff
@@ -545,3 +545,166 @@ func TestGetRepositoryByUUID(t *testing.T) {
...
+			Name:           "should create repository successfully",
+			RepoToCreate:   repoToCreate,
+			HTTPStatus:     http.StatusCreated,
+			IOReadAll:      io.ReadAll,
+			NewJSONEncoder: json.NewEncoder,
+			Response:       &responseRepo,
+			ExpectedError:  nil,
+		},
+		{
+			Name:           "should return error when http status is not 201",
```
Comment (mgold1234) [inline pkg/clients/repositories/clients_test.go:593]: do you know what is the status error you get? 
maybe specify it in the Name test?

Comment (mgold1234) [reply pkg/clients/repositories/clients_test.go:593]: at this specific test you check StatusBadRequest                   = 400, so we can define it in the test name as well

Comment (ldjebran) [reply pkg/clients/repositories/clients_test.go:593]: The most important in this test is that the returned status is not 201, I can put 500 as returned status and the result will be the same, so the most important here is not what we will receive as returned status, but how our code will react when the status is different from 201.

Comment (ldjebran) [reply pkg/clients/repositories/clients_test.go:593]: Also you can read in the code that we are checking if the returned http status is different from 201, and not to equal to 400.

**Review by mgold1234 — APPROVED** [review]
/lgtm

**Pull request merged**